lib/python/picongpu/picmi/diagnostics/unit_dimension.py

```diff
@@ -0,0 +1,99 @@
+"""
+This file is part of PIConGPU.
+Copyright 2025 PIConGPU contributors
+Authors: Richard Pausch, Julian Lenz
+License: GPLv3+
+"""
+
+import numpy as np
+
+_EPSILON = 1.0e-6
+
+
+class UnitDimension:
+    """
+    class to describe units
+    """
+
+    # number of unit dimension
+    N_unit_dim = 7
+    unit_vector = np.zeros(N_unit_dim)
+
+    """
+    The unit index map associates names with array indices.
+    The name definition can be found at https://en.wikipedia.org/wiki/SI_base_unit.
+    The array index order follows that of PIConGPU:
+    include/picongpu/plugins/binning/UnitConversion.hpp, lines 40-48.
+    """
+    _unit_index_map = {
+        "length": 0,
+        "L": 0,
+        "mass": 1,
+        "M": 1,
+        "time": 2,
+        "T": 2,
+        "electric current": 3,
+        "I": 3,
+        "thermodynamic temperature": 4,
+        "Θ": 4,
```

Comment:
not a fan of a non ASCII character even though that is the SI-suggestion

```suggestion
        "K": 4,
```

Reply:
We had the discussion on nomenclature changes between publication and code is the root of a lot of confusion. 
Thus, I would stick with the standard nomenclature Θ.

Reply:
I am not a fan of non-ASCII characters. In general, my question would go more in the direction of why we do not use a unit system library?
Do we not need unit conversion?

Reply:
Regarding the last question: yes, otherwise we get unusable numbers out. 
Regarding the use of a unit system library: yes, there are options like `pint` which we might use in the future. But for now, they are a overkill. 
I agree with the non-ASCII characters, a strongly disagree with using `T`. @psychocoderHPC would you be fine to fix this in a small follow up PR?